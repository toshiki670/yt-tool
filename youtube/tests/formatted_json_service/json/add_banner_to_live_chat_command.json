{
  "replayChatItemAction": {
    "actions": [
      {
        "addBannerToLiveChatCommand": {
          "bannerRenderer": {
            "liveChatBannerRenderer": {
              "header": {
                "liveChatBannerHeaderRenderer": {
                  "icon": { "iconType": "KEEP" },
                  "text": {
                    "runs": [
                      { "text": "Pinned by " },
                      {
                        "text": "チャンネル名"
                      }
                    ]
                  },
                  "contextMenuButton": {
                    "buttonRenderer": {
                      "icon": { "iconType": "MORE_VERT" },
                      "accessibility": { "label": "Chat actions" },
                      "accessibilityData": {
                        "accessibilityData": { "label": "Chat actions" }
                      },
                      "command": {
                        "commandMetadata": {
                          "webCommandMetadata": { "ignoreNavigation": true }
                        },
                        "liveChatItemContextMenuEndpoint": {
                          "params": "Q2g0S0hBb2FRMGt5YVY5eFlXWTBiMmRFUmxKbVIzZG5VV1JLYkhOaFYzY2FLU29uQ2hoVlF6WnZURGhuWlV0TlpYbG5iUzAwT1dsTWNuRmxabmNTQzI4d1QyaDVWVXMwZDJseklBRW9CRElhQ2hoVlF6WnZURGhuWlV0TlpYbG5iUzAwT1dsTWNuRmxabmM0QVVnQVVBRSUzRA=="
                        }
                      }
                    }
                  }
                }
              },
              "contents": {
                "liveChatTextMessageRenderer": {
                  "message": {
                    "runs": [
                      {
                        "text": "https://www.youtube.com/watch?v=OGdjV...",
                        "navigationEndpoint": {
                          "commandMetadata": {
                            "webCommandMetadata": {
                              "url": "https://www.youtube.com/watch?v=OGdjVWod-rs",
                              "webPageType": "WEB_PAGE_TYPE_UNKNOWN",
                              "rootVe": 83769
                            }
                          },
                          "urlEndpoint": {
                            "url": "https://www.youtube.com/watch?v=OGdjVWod-rs",
                            "target": "TARGET_NEW_WINDOW",
                            "nofollow": true
                          }
                        }
                      }
                    ]
                  },
                  "authorName": {
                    "simpleText": "チャンネル名"
                  },
                  "authorPhoto": {
                    "thumbnails": [
                      {
                        "url": "https://yt4.ggpht.com/JvnR_Vv_2eMqmKEQ-_-Rqbm6JcwJmawPy8jFU60QYbLxeLNSHUWIenI-JxQDPD5j-8QTV0bC16M=s32-c-k-c0x00ffffff-no-rj",
                        "width": 32,
                        "height": 32
                      },
                      {
                        "url": "https://yt4.ggpht.com/JvnR_Vv_2eMqmKEQ-_-Rqbm6JcwJmawPy8jFU60QYbLxeLNSHUWIenI-JxQDPD5j-8QTV0bC16M=s64-c-k-c0x00ffffff-no-rj",
                        "width": 64,
                        "height": 64
                      }
                    ]
                  },
                  "id": "ChwKGkNJMmlfcWFmNG9nREZSZkd3Z1FkSmxzYVd3",
                  "timestampUsec": "1727409889983467",
                  "authorBadges": [
                    {
                      "liveChatAuthorBadgeRenderer": {
                        "icon": { "iconType": "OWNER" },
                        "tooltip": "Owner",
                        "accessibility": {
                          "accessibilityData": { "label": "Owner" }
                        }
                      }
                    }
                  ],
                  "authorExternalChannelId": "UC6oL8geKMeygm-49iLrqefw"
                }
              },
              "actionId": "ChwKGkNQejlyYW1mNG9nREZSWU5yUVlkZEVZS0x3",
              "viewerIsCreator": false,
              "targetId": "live-chat-banner",
              "isStackable": false,
              "backgroundType": "LIVE_CHAT_BANNER_BACKGROUND_TYPE_STATIC",
              "bannerProperties": { "autoCollapseDelay": { "seconds": "7" } },
              "bannerType": "LIVE_CHAT_BANNER_TYPE_PINNED_MESSAGE"
            }
          }
        }
      }
    ]
  },
  "videoOffsetTimeMsec": "14941793",
  "isLive": true
}
