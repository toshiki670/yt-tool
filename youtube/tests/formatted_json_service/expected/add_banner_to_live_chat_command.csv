id,authorExternalChannelId,postedAt,category,authorName,content
ChwKGkNQejlyYW1mNG9nREZSWU5yUVlkZEVZS0x3,,2024-09-27T13:04:49.983467+09:00,Banner,Pinned by チャンネル名,TextMessageID: 'ChwKGkNJMmlfcWFmNG9nREZSZkd3Z1FkSmxzYVd3'
ChwKGkNJMmlfcWFmNG9nREZSZkd3Z1FkSmxzYVd3,UC6oL8geKMeygm-49iLrqefw,2024-09-27T13:04:49.983467+09:00,Text message,チャンネル名,message: 'https://www.youtube.com/watch?v=OGdjVWod-rs'
